plugin/src/main/java/org/gradle/testretry/internal/executer/RetryTestResultProcessor.java

```diff
@@ -97,9 +97,18 @@ public void completed(Object testId, TestCompleteEvent testCompleteEvent) {
                     addRetry(className, name);
                 }
 
+                // test class lifecycle which was successful but failed on retry can prevent tests from being re-tried
+                // in this case, retry all non-retried tests from previous run
+                if (isLifecycleFailure(className, name) && !failedInPreviousRound) {
+                    previousRoundFailedTests.remove(className, n -> {
```

Comment:
💬 Actually, it would be enough to clear all failed tests for the entire class, since this lifecycle failure will trigger the retry of the entire class. But I think re-adding failed methods for retry is more semantically correct, and if those will not be retried for some other reason, we would still throw the same error.